src/generators/mappersGenerator.ts

```diff
@@ -21,29 +28,132 @@ export function generateMappers(
     moduleSpecifier: "@azure/core-http"
   });
 
-  for (const mapper of clientDetails.mappers) {
-    // Generate the mapper definition and replace object key strings
-    // with the unquoted name
-    const mapperString = JSON.stringify(mapper, undefined, 2).replace(
-      /\"([^"]+)\":/g,
-      "$1:"
-    );
+  let discriminators: { [key: string]: string } = {};
 
+  for (const mapper of clientDetails.mappers) {
     if (mapper.type.name === MapperType.Composite && mapper.serializedName) {
+      const compositeMapper = mapper as CompositeMapper;
+
+      if (compositeMapper.type.polymorphicDiscriminator) {
+        const { uberParent, className } = compositeMapper.type;
+        if (!uberParent || !className) {
+          throw new Error(
+            `Expected CompositeMapper with polymorphicDiscriminator to specify uberParent property uberParent: ${uberParent} className: ${className}`
+          );
+        }
+
+        if (uberParent === className) {
+          discriminators[className] = className;
+        } else {
+          discriminators[`${uberParent}.${mapper.serializedName}`] = className;
+        }
+      }
+
       mappersFile.addVariableStatement({
         isExported: true,
         declarations: [
           {
-            name: normalizeName(mapper.serializedName, NameType.Class),
+            name: normalizeName(
+              compositeMapper.type.className || "",
```

Comment:
What happens when name is assigned to an empty string?

Reply:
I have added a check to throw an exception if no ClassName was found